src/form.svelte

```diff
@@ -19,6 +19,14 @@
     titlePlaceholder = tabInfo.title;
     const availableTags = await getTags().catch(() => [])
     availableTagNames = availableTags.map(tag => tag.name)
+
+    const bookmarksFound = await search(url, { limit: 1000 }).catch(() => [])
```

Comment:
Loading 1000 bookmarks is overkill here, and there are users with more than 1000 bookmarks. Instead, the REST API also has a `bookmarks/check` endpoint, that allows to check if a bookmark for a URL already exists. An example of it being used can be found in the add bookmark form of the linkding application itself: https://github.com/sissbruecker/linkding/blob/master/bookmarks/templates/bookmarks/form.html#L123.
For a proper implementation we should add a `check` function to the `linkding.js` JS file that takes an URL, executes the API request and returns the check result. Then call that function from here.

Reply:
Yes of course, in this case the limit is actually not the number of bookmarks returned. The search returns very few results as they need to match the query which is the URL entered.
I used 1000 to basically ignore the limit but I can see how it is misleading without a comment to explain it

I will look into the other form, thank you for the reference

Reply:
Ah, I did not notice the search param. I also saw that the `bookmarks/check` response only returns the bookmark ID, so with that approach there would need to be a second request to retrieve the actual bookmark data (using `bookmarks/{id}`. Still, that feels like the more correct approach.

Reply:
I agree that the search isn't a very good option as it feels like a hack haha

I looked at the `bookmarks/check` implementation and I am a bit worried about the potential performance impact of the call to `load_website_metadata` which isn't required in this particular context.
It would mean that we effectively load the entire website each time we check for a URL (which could be as often as the user types)

How would you approach this problem? 
My first instinct would be to either split the endpoint or to parameterize it but I am not familiar with your conventions

Reply:
That's a good point, but I don't think it's something that we need to worry about. The in-app form works basically the same way, and there haven't been any complaints or issues so far. The check API will definitely be slower than the search API, however we do not degrade any existing features of the extension. Also the way you implemented it, the URL will actually only be checked once on initialization, but not when modifying it. I think that's probably good enough for now, as users will rarely modify the URL when bookmarking a page from the extension.

Long-term loading the website metadata could actually be beneficial in another way, in that it would allow to prefill the input fields with the scraped metadata, which would replicate the behavior of the in-application form. The current situation is a bit inconsistent:
- the in-app form uses scraped metadata to prefill title and description (technically the inputs placeholders)
- the extension form uses the tab's title to prefill the title, but does not prefill the description
- the prefilled values are actually never sent to the backend, they are just a visual indicator of how the bookmark will look like if you don't enter custom value. When you save a bookmark, the backend will always fetch the website metadata (again) and save the scraped values. That means there can be inconsistencies if the scraper detects a different title than the extension.

So, I still think using the existing check API is fine, and as mentioned above it would be a good first step in aligning the behaviour of the two forms.

Reply:
Thank you for the detailed answer! 
I'll go with this solution as soon as I have some time to work on it